packages/app-accounts/src/UploadButton.tsx

```diff
@@ -0,0 +1,324 @@
+                    </div>
+                    <div className='expanded'>
+                      <p>
+                        <Trans i18nKey='unlock.info'>
+                          Please enter password for account <span className='code'>{shortValue}</span> to upload it encrypted.
+                        </Trans>
+                      </p>
+                    </div>
+                    {this.renderContent()}
+                  </div>
+                  {this.renderButtons()}
+                </div>
+              </Modal.Content>
+            </Modal>
+          ) : null
+        }
+        <ReactFileReader fileTypes={['.json']} base64={false} multipleFiles={false} handleFiles={this.handleUploadedFiles}>
+          <Button
+            className={className}
+            icon={icon}
+            isCircular={isCircular}
+            isPrimary={isPrimary}
+            size={size}
+            style={style}
+          />
+        </ReactFileReader>
+      </div>
+    );
+  }
+
+  renderContent () {
+    const { uploadedFileKeyringPair } = this.state;
+
+    if (isUndefined(uploadedFileKeyringPair) || !uploadedFileKeyringPair.address) {
+      return null;
+    }
+
+    return (
+      <div>
+        { this.renderUnlock() }
+      </div>
+    );
+  }
+
+  renderUnlock () {
+    const { t } = this.props;
+    const { uploadedFileKeyringPair, password, unlockError } = this.state;
+    let translateError = null;
+
+    if (isUndefined(uploadedFileKeyringPair) || !uploadedFileKeyringPair.address) {
+      return null;
+    }
+
+    const keyringAddress: KeyringAddress = keyring.getAddress(uploadedFileKeyringPair.address);
+
+    if (unlockError && unlockError.key && unlockError.value) {
+      translateError = {
+        key: t(unlockError.key),
+        value: t(unlockError.value)
+      };
+    }
+
+    return (
+      <Unlock
+        autoFocus={true}
+        error={translateError || null}
+        onChange={this.onChangePassword}
+        password={password}
+        passwordWidth={'full'}
+        value={keyringAddress.publicKey()}
+      />
+    );
+  }
+
+  renderButtons () {
+    const { t } = this.props;
+
+    return (
+      <Modal.Actions>
+        <Button.Group>
+          <Button
+            isNegative
+            onClick={this.onDiscard}
+            text={t('creator.discard', {
+              defaultValue: 'Cancel'
+            })}
+          />
+          <Button.Or />
+          <Button
+            isDisabled={false}
+            isPrimary
+            onClick={this.onSubmit}
+            text={t('creator.submit', {
+              defaultValue: 'Submit'
+            })}
+          />
+        </Button.Group>
+      </Modal.Actions>
+    );
+  }
+
+  emptyState (): State {
+    return {
+      address: '',
+      password: '',
+      isPasswordModalOpen: false,
+      unlockError: null,
+      uploadedFileKeyringPair: undefined
+    };
+  }
+
+  nextState (newState: State): void {
+    this.setState(
```

Comment:
As per the other vomponent, is there a specific reason for `nextState`, does not look like any state mutation takes place. `setState` will do fine and remove extra noise.